src/components/test/QuickOpenModal.spec.js

```diff
@@ -210,6 +287,542 @@ describe("QuickOpenModal", () => {
+        results: [{}, { id }],
+        selectedIndex: 1
+      }));
+      const event = {
+        key: "Enter"
+      };
+      wrapper.find("SearchInput").simulate("keydown", event);
+      expect(props.selectLocation).not.toHaveBeenCalled();
+      expect(props.setQuickOpenQuery).toHaveBeenCalledWith(id);
+    });
+
+    it("on Tab, close modal", () => {
+      const { wrapper, props } = generateModal(
+        {
+          enabled: true,
+          query: ":34:12",
+          searchType: "goto"
+        },
+        "shallow"
+      );
+      const event = {
+        key: "Tab"
+      };
+      wrapper.find("SearchInput").simulate("keydown", event);
+      expect(props.closeQuickOpen).toHaveBeenCalled();
+      expect(props.selectLocation).not.toHaveBeenCalled();
+    });
+
+    it("on ArrowUp, traverse results up with functions", () => {
+      const sourceId = "sourceId";
+      const { wrapper, props } = generateModal(
+        {
+          enabled: true,
+          query: "test",
+          searchType: "functions",
+          selectedSource: I.Map({ id: sourceId }),
+          symbols: {
+            functions: [{ title: "functionA" }, { title: "anonymous" }],
+            variables: {}
+          }
+        },
+        "shallow"
+      );
+      const event = {
+        preventDefault: jest.fn(),
+        key: "ArrowUp"
+      };
+      const location = {
+        start: {
+          line: 1
+        },
+        end: {
+          line: 3
+        }
+      };
+      wrapper.setState(() => ({
+        results: [{ id: "0", location }, { id: "1" }, { id: "2" }],
+        selectedIndex: 1
+      }));
+      wrapper.find("SearchInput").simulate("keydown", event);
+      expect(event.preventDefault).toHaveBeenCalled();
+      expect(wrapper.state().selectedIndex).toEqual(0);
+      expect(props.highlightLineRange).toHaveBeenCalledWith({
+        end: 3,
+        sourceId: sourceId,
+        start: 1
+      });
+    });
+
+    it("on ArrowDown, traverse down with variables", () => {
+      const sourceId = "sourceId";
+      const { wrapper, props } = generateModal(
+        {
+          enabled: true,
+          query: "test",
+          searchType: "variables",
+          selectedSource: I.Map({ id: sourceId }),
+          symbols: {
+            functions: [{ title: "functionA" }, { title: "anonymous" }],
+            variables: {}
+          }
+        },
+        "shallow"
+      );
+      const event = {
+        preventDefault: jest.fn(),
+        key: "ArrowDown"
+      };
+      const location = {
+        start: {
+          line: 7
+        }
+      };
+      wrapper.setState(() => ({
+        results: [{ id: "0", location }, { id: "1" }, { id: "2" }],
+        selectedIndex: 1
+      }));
+      wrapper.find("SearchInput").simulate("keydown", event);
+      expect(event.preventDefault).toHaveBeenCalled();
+      expect(wrapper.state().selectedIndex).toEqual(2);
+      expect(props.selectLocation).toHaveBeenCalledWith({
```

Comment:
these results are bit confusing, why are we at line 0? is it taking us to the top of the file on `arrowDown` within the variables search?

Reply:
ya, so it's opening the file with that variable, but that variable does not have a location, so it defaults to line 0

Reply:
that might be worth putting into a separate test, because here the test is expecting variable traversal and what we are testing is if the default behavior is what we expect

Reply:
done